Sources/TensorFlow/Core/Utilities.swift

```diff
@@ -167,3 +167,157 @@ extension UnsafeMutablePointer where Pointee == CTensorHandle? {
         self.init(unwrapped)
     }
 }
+
+//===------------------------------------------------------------------------------------------===//
+// Hashing
+//===------------------------------------------------------------------------------------------===//
+
+internal extension FixedWidthInteger {
+    init(bytes: ArraySlice<UInt8>, startingAt index: Int) {
+        if bytes.isEmpty { self.init(0); return }
+        let count = bytes.count
+        self.init(0)
+        for i in 0..<MemoryLayout<Self>.size {
+            let j = (MemoryLayout<Self>.size - i - 1) * 8
+            self |= count > 0 ? Self(bytes[index.advanced(by: i)]) << j : 0
+        }
+    }
+
+    func bytes(count byteCount: Int = MemoryLayout<Self>.size) -> [UInt8] {
+        var littleEndianValue = littleEndian
+        return withUnsafePointer(to: &littleEndianValue) { pointer -> [UInt8] in
+            let bytesPointer = UnsafeMutablePointer<UInt8>(OpaquePointer(pointer))
+            var bytes = [UInt8](repeating: 0, count: byteCount)
+            for i in 0..<Swift.min(MemoryLayout<Self>.size, byteCount) {
+                bytes[byteCount - 1 - i] = (bytesPointer + i).pointee
+            }
+            return bytes
+        }
+    }
+
+    fileprivate func rotate(rightBy count: Self) -> Self {
```

Comment:
If a method's phrase has the syntactic structure `(VP V Adv (PP P NP))`, it is more common to put the adverb in the base name than in the argument label with the prepositional phrase.

```suggestion
    fileprivate func rotateRight(by count: Self) -> Self {
```

Reply:
I'm actually working on removing the need for rotations now to better utilize SIMD. :)